packages/shared-lib/src/types/utility.ts

```diff
@@ -5,3 +5,4 @@ export type DeepValueOf<T> = T extends object
   : T;
 
 export type NonEmptyArray<T> = [T, ...T[]];
+export type NonEmptyArrayReadOnly<T> = readonly [T, ...T[]];
```

Comment:
```suggestion
export type NonEmptyReadonlyArray<T> = readonly [T, ...T[]];
```
`ReadonlyArray` 타입과 컨벤션을 맞춰야할 것 같습니다

Reply:
그리고 개인적으로, `Readonly<NonEmptyArray<T>>`보다 가독성이 높은지는 잘 모르겠습니다,,